File: javascript/resistor-color-trio/resistor-color-trio.js
const colorValues = new Map([
  ['black', 0], 
  ['brown', 1], 
  ['red', 2], 
  ['orange', 3], 
  ['yellow', 4], 
  ['green', 5], 
  ['blue', 6], 
  ['violet', 7], 
  ['grey', 8], 
  ['white', 9]
]);

export class ResistorColorTrio {
  constructor(colors) {
    let value = (10 * colorValues.get(colors[0]) + colorValues.get(colors[1])) * (10 ** colorValues.get(colors[2]));
    if (isNaN(value)) throw new Error('invalid color');
    this.value = value;
  }

  get label() {
    if (this.value >= 1000) {
      return `Resistor value: ${this.value/1000} kiloohms`;
    } else {
      return `Resistor value: ${this.value} ohms`;
    }
  }
}

File: go/resistor-color-trio/resistor_color_trio.go
package resistorcolortrio

import (
    "math"
    "strconv"
)

// Label describes the resistance value given the colors of a resistor.
func Label(colors []string) string {
	colorMap := map[string]int{"black": 0, "brown": 1, "red": 2, "orange": 3, "yellow": 4, "green": 5, "blue": 6, "violet": 7, "grey": 8, "white": 9}
    ohms := (10 * colorMap[colors[0]] + colorMap[colors[1]]) * int(math.Pow10(colorMap[colors[2]]))
    if ohms >= 1000000000 {
        return strconv.Itoa(ohms/1000000000) + " gigaohms"
    } else if ohms >= 1000000 {
    	return strconv.Itoa(ohms/1000000) + " megaohms"
    } else if ohms >= 1000 {
    	return strconv.Itoa(ohms/1000) + " kiloohms"
    } else {
    	return strconv.Itoa(ohms) + " ohms"
    }
}
